src/Tokenizer/Tokens.php

```diff
@@ -1396,7 +1396,11 @@ private function changeCodeHash($codeHash)
      */
     private function registerFoundToken($token)
     {
-        $tokenKind = $this->extractTokenKind($token);
+        // inlined extractTokenKind() call on the hot path
+        $tokenKind = $token instanceof Token
+            ? ($token->isArray() ? $token->getId() : $token->getContent())
+            : (\is_array($token) ? $token[0] : $token)
+        ;
```

Comment:
on big fan of this.
DRY is violated. Tokenizer classes are heart of PHP CS Fixer, having code copy-pasted is not helping with a maintenance :(